### DISCUSSION TITLE
Build pattern?

### DISCUSSION THREAD

**Discussion by Brian-McBride** [swordev/suid#69] [General]
First, this is a pretty cool project. 
It makes me wonder if it would be possible to automate a port of MUI to 100% Web Component standards so that it can just work in any dom reactive lib?

 I was wondering if you can briefly explain the automation that you are using here? I'm gathering that you are downloading MUI 5.5.0 based on the scripts. Are you just downloading it and then running a patch across them all, sort of like what I've seen with the React2Solid playground in your docs? 

One of the reasons I don't approach Solid for projects is the lack of a well-developed UX Component library. Unfortunately, speed and budget are things I have to worry about and the SolidJS ecosystem is getting better, but lacks. An automated port of MUI to SolidJS that could keep up with the MUI release versions would sure help my teams grab it as a tool easier.

**Comment by juanrgm** [comment]
> It makes me wonder if it would be possible to automate a port of MUI to 100% Web Component standards so that it can just work in any dom reactive lib?

Currently is not possible to transform all MUI components at 100% automatically. Each component has its quirks. There are components where the transform is excellent (100%), but there are other componentes (ex: TextField) where the transform reachs the ~75%.

Even so, first it is neccesary to have the core available (`@suid/types`, `@suid/utils`, `@suid/css`, `@suid/system`) for supporting the MUI components, and those packages were ported/created manually.

> I was wondering if you can briefly explain the automation that you are using here? I'm gathering that you are downloading MUI 5.5.0 based on the scripts. Are you just downloading it and then running a patch across them all, sort of like what I've seen with the React2Solid playground in your docs?

The tool is under development (it is not in the repository) but I will release it soon.
